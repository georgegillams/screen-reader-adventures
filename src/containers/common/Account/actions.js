import { createRoutine } from 'redux-saga-routines';

import { LOGOUT, REQUEST_VERIFICATION_EMAIL } from './constants';

export const logout = createRoutine(LOGOUT);
export const requestVerificationEmail = createRoutine(REQUEST_VERIFICATION_EMAIL);
