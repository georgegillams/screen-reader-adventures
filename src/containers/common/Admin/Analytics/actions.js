import { createRoutine } from 'redux-saga-routines';

import { LOAD } from './constants';

export const load = createRoutine(LOAD);
