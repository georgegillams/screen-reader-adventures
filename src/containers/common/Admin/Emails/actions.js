import { createRoutine } from 'redux-saga-routines';

import { LOAD, RESEND } from './constants';

export const load = createRoutine(LOAD);
export const resend = createRoutine(RESEND);
