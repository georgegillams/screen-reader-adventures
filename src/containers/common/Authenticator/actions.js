import { createRoutine } from 'redux-saga-routines';

import { LOAD_AUTH, SET_USER } from './constants';

export const loadAuth = createRoutine(LOAD_AUTH);
export const setUser = createRoutine(SET_USER);
