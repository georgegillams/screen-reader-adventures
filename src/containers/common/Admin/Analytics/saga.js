import { takeLatest, put, call } from 'redux-saga/effects';

import request from 'client-utils/common/request';
import apiStructure from 'helpers/common/apiStructure';

import { load } from './actions';

export function* doLoad() {
  const requestURL = apiStructure.loadAnalyticsSummary.fullPath;

  try {
    yield put(load.request());

    const result = yield call(request, requestURL);

    if (result.error) {
      yield put(load.failure(result));
    } else {
      yield put(load.success(result.analytics));
    }
  } catch (err) {
    yield put(load.failure(err));
  } finally {
    yield put(load.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(load.TRIGGER, doLoad);
}
