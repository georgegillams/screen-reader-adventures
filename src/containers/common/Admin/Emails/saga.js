import { takeLatest, put, call, select } from 'redux-saga/effects';

import request from 'client-utils/common/request';
import apiStructure from 'helpers/common/apiStructure';

import { load, resend } from './actions';
import { selectState } from './selectors';

export function* doLoad() {
  const requestURL = apiStructure.loadEmails.fullPath;

  try {
    yield put(load.request());

    const result = yield call(request, requestURL);

    if (result.error) {
      yield put(load.failure(result));
    } else {
      yield put(load.success(result.emails));
    }
  } catch (err) {
    yield put(load.failure(err));
  } finally {
    yield put(load.fulfill());
  }
}

export function* doResend() {
  const currentState = yield select(selectState());
  const { emailToResend } = currentState;
  const requestURL = apiStructure.resendEmail.fullPath;

  try {
    yield put(resend.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
      body: JSON.stringify(emailToResend),
      headers: {
        'Content-Type': 'application/json',
      },
    });

    if (result.error) {
      yield put(resend.failure(result));
    } else {
      yield put(resend.success(result.emails));
      yield put(load.trigger());
    }
  } catch (err) {
    yield put(resend.failure(err));
  } finally {
    yield put(resend.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(load.TRIGGER, doLoad);
  yield takeLatest(resend.TRIGGER, doResend);
}
