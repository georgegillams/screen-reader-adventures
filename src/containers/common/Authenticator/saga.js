import { takeLatest, put, call } from 'redux-saga/effects';

import request from 'client-utils/common/request';
import apiStructure from 'helpers/common/apiStructure';

import { loadAuth } from './actions';

export function* doLoadAuth() {
  const requestURL = apiStructure.loadAuth.fullPath;

  try {
    yield put(loadAuth.request());

    const result = yield call(request, requestURL);

    yield put(loadAuth.success(result));
  } catch (err) {
    yield put(loadAuth.failure(err));
  } finally {
    yield put(loadAuth.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(loadAuth.TRIGGER, doLoadAuth);
}
