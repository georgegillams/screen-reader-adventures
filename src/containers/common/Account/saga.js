import { takeLatest, put, call } from 'redux-saga/effects';

import request from 'client-utils/common/request';
import apiStructure from 'helpers/common/apiStructure';

import { logout, requestVerificationEmail } from './actions';
import { setUser } from 'containers/common/Authenticator/actions';

export function* doLogout() {
  const requestURL = apiStructure.logout.fullPath;

  try {
    yield put(logout.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
    });

    if (result.error) {
      yield put(logout.failure(result));
    } else {
      yield put(logout.success(result));
      yield put(setUser.trigger(null));
    }
  } catch (err) {
    yield put(logout.failure(err));
  } finally {
    yield put(logout.fulfill());
  }
}

export function* doRequestVerificationEmail() {
  const requestURL = apiStructure.requestVerificationEmail.fullPath;

  try {
    yield put(requestVerificationEmail.request());

    const result = yield call(request, requestURL, {
      method: 'POST',
    });

    if (result.error) {
      yield put(requestVerificationEmail.failure(result));
    } else {
      yield put(requestVerificationEmail.success(result));
    }
  } catch (err) {
    yield put(requestVerificationEmail.failure(err));
  } finally {
    yield put(requestVerificationEmail.fulfill());
  }
}

export default function* saga() {
  yield takeLatest(logout.TRIGGER, doLogout);
  yield takeLatest(requestVerificationEmail.TRIGGER, doRequestVerificationEmail);
}
