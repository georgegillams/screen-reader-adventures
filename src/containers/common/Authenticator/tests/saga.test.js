import { put, takeLatest } from 'redux-saga/effects';

import { loadAuth } from '../actions';

import saga, { doLoadAuth } from '../saga';

describe('Authenticator saga', () => {
  let mainSaga;

  beforeEach(() => {
    mainSaga = saga();
  });

  it('Should get authenticated user on TRIGGER', () => {
    const takeLatestDescriptor = mainSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeLatest(loadAuth.TRIGGER, doLoadAuth));
  });

  describe('loadAuthenticator actions', () => {
    let loadAuthGenerator;

    const response = {
      name: 'userName',
      status: 200,
    };

    beforeEach(() => {
      loadAuthGenerator = doLoadAuth();

      const selectDescriptor = loadAuthGenerator.next().value;
      expect(selectDescriptor).toMatchSnapshot();
    });

    it('Should call loadAuth.success on successful API call', () => {
      loadAuthGenerator.next();
      const putSuccess = loadAuthGenerator.next(response).value;
      loadAuthGenerator.next();

      expect(putSuccess).toEqual(put(loadAuth.success(response)));
    });

    it('Should call loadAuth.failure if an exception occurs', () => {
      const response = new Error('Some error');
      const putFailure = loadAuthGenerator.throw(response).value;

      expect(putFailure).toEqual(put(loadAuth.failure(response)));
    });
  });
});
