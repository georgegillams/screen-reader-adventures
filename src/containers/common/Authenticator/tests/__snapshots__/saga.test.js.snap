// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authenticator saga loadAuthenticator actions Should call loadAuth.failure if an exception occurs 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/authenticator/LOAD_AUTH/REQUEST",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`Authenticator saga loadAuthenticator actions Should call loadAuth.success on successful API call 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "app/authenticator/LOAD_AUTH/REQUEST",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;
